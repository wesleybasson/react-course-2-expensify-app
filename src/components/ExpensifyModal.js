import React from 'react';
import Modal from 'react-modal';

export default (props) => (
    <Modal
        isOpen={props.modalOpen}
        onRequestClose={props.handleClose}
        contentLabel="Expensify"
        closeTimeoutMS={200}
        className="modal"
        ariaHideApp={false}
    >
        <h3 className="modal__title">{props.title}</h3>
        <p className="modal__body">Do you really want to remove this item?</p>
        <div className="container">
            <button className="button button__half button--success" onClick={props.handleAction}>Yes</button>
            <button className="button button__half button--cancel" onClick={props.handleClose}>No</button>
        </div>
    </Modal>
);