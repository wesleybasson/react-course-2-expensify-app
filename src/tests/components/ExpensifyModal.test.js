import React from 'react';
import { shallow } from 'enzyme';
import ExpensifyModal from '../../components/ExpensifyModal';

let wrapper, handleClose, handleAction;

beforeEach(() => {
    handleClose = jest.fn();
    handleAction = jest.fn();
    wrapper = shallow(<ExpensifyModal handleClose={handleClose} handleAction={handleAction} />)
});

test('should render ExpensifyModal correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle modal action', () => {
    wrapper.find('button').at(0).simulate('click');
    expect(handleAction).toHaveBeenCalled();
});

test('should handle closing of modal', () => {
    wrapper.find('button').at(1).simulate('click');
    expect(handleClose).toHaveBeenCalled();
});