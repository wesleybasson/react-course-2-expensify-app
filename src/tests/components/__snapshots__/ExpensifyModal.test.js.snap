// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensifyModal correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Expensify"
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="modal__title"
  />
  <p
    className="modal__body"
  >
    Do you really want to remove this expense?
  </p>
  <div
    className="container"
  >
    <button
      className="button button__half button--success"
      onClick={[Function]}
    >
      Yes
    </button>
    <button
      className="button button__half button--cancel"
      onClick={[Function]}
    >
      No
    </button>
  </div>
</Modal>
`;
